frontend/src/stores/auth.js

```diff
@@ -59,11 +58,21 @@ export const useAuthStore = defineStore('auth', {
     async setUserInfo(userInf) {
       if (userInf) {
         this.userInfo = userInf
+        this.userHasRoles = userInf.roles && userInf.roles.length > 0
       } else {
         this.userInfo = null
       }
```

Comment:
I'm not sure I'd bother with a setter here. I'd just update the state directly and make userHasRoles a getter (the equivalent of computed) instead of a part of the state.

Reply:
Confirming completed.